src/sentry/integrations/slack/webhooks/event.py

```diff
@@ -116,7 +118,25 @@ def prompt_link(self, slack_request: SlackDMRequest) -> None:
                 text=payload["text"],
                 **SlackPromptLinkMessageBuilder(associate_url).as_payload(),
             )
-        except SlackApiError:
+        except SlackApiError as e:
+            # If the channel is not found, it could be because the bot doesn't have access to the channel to send an ephemeral message
+            # In this case, we should send the message to the user in the bot <> user DM
+            if unpack_slack_api_error(e) == CHANNEL_NOT_FOUND:
```

Comment:
Do other messages other than link your account go through this logic? If so, what if the channel is actually not found because it's deleted or something? Would it still make sense to get a DM?

Reply:
link your account is the only flow that goes through this logic. the only other cases we would get a CHANNEL_NOT_FOUND error is if the channel was deleted between us getting the unfurl webhook and sending the message and the channel id slack sent to us being valid, both extremely unlikely.